app/services/reshare_service.rb

```diff
@@ -7,7 +7,6 @@ def initialize(user=nil)
 
   def create(post_id)
     post = post_service.find!(post_id)
-    raise I18n.t("reshares.create.error") if user.guid == post.author.guid
```

Comment:
Why do you remove this check? I saw you put it elsewhere. But it seems ok to verify it on all creation, wether it comes from an helper or not.

Reply:
This was my first cut implementation since I saw it on the service layer but when I posted about this bug (someone can reshare their own post).  I posted about the problem and the solution I made on the IRC channel and it was suggested it move down to the model/data layer instead of the service layer so that's where I put it.  You are right that it couldn't hurt to check it in two places even if it is a bit redundant but I'm not sure it is necessary either.